crates/symbolicator-service/src/services/sourcemap_lookup.rs

```diff
@@ -402,17 +404,23 @@ impl ArtifactFetcher {
         source: ByteViewString,
         sourcemap: ByteViewString,
     ) -> CacheEntry<OwnedSourceMapCache> {
-        // TODO: really hook this up to the `Cacher`.
-        // this is currently blocked on figuring out combined cache keys that depend on both
-        // `source` and `sourcemap`.
-        // For the time being, this all happens in a temp file that we throw away afterwards.
-        let req = FetchSourceMapCacheInternal { source, sourcemap };
-
-        let mut temp_file = self.sourcemap_caches.tempfile()?;
-        req.compute(&mut temp_file).await?;
+        let cache_key = {
+            let mut cache_key = CacheKey::scoped_builder(&self.scope);
+            let source_hash = Sha256::digest(&source);
+            let source_hash = BASE64.encode(&source_hash);
+            let sourcemap_hash = Sha256::digest(&sourcemap);
+            let sourcemap_hash = BASE64.encode(&sourcemap_hash);
+
+            cache_key.write_str("source:\n").unwrap();
+            cache_key.write_str(&source_hash).unwrap();
+            cache_key.write_str("\nsourcemap:\n").unwrap();
+            cache_key.write_str(&sourcemap_hash).unwrap();
+
```

Comment:
Would it be possible to write it like so? Or is there a reason you split it by section?

```rust
let cache_key_value = format!("source:\n{source_hash}\nsourcemap:\n{sourcemap_hash}");
cache_key.write_str(cache_key_value).unwrap();
```

Reply:
I left it as two `write`s now, hope that's ok.